devel/rust-analyzer/Portfile

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cargo 1.0
+
+github.setup        rust-analyzer rust-analyzer 2021-01-18 v
+categories          devel
+platforms           darwin
+maintainers         @fabius openmaintainer
+license             Apache-2
+
+homepage            https://rust-analyzer.github.io/
+description         An experimental Rust \
+                    compiler front-end for IDEs
+
+long_description    rust-analyzer is an experimental modular \
+                    compiler frontend for the Rust language. \
+                    It is a part of a larger rls-2.0 effort \
+                    to create excellent IDE support for Rust.
+
+master_sites        https://github.com/rust-analyzer/rust-analyzer/archive/
+distname            ${version}
+checksums           rmd160  c0db3ee3e3f6db7a03f8b53fa45b180b760f6504 \
+                    sha256  009ab946848719bb05bcaddab1cf5a75a45c5a92ae6dc31763adb42c3229f7db \
+                    size    1435696
+supported_archs     x86_64 arm64
+
+build {
+    system -W ${worksrcpath} "rm lib/README.md"
+    system -W ${worksrcpath} "cargo build --release"
```

Comment:
The `cargo` portgroup (`PortGroup cargo 1.0`) normally does the `cargo build` step for you, which means after specifying it, you don't have to specify an explicit `build` section.  Did the normal structure not work for a particular reason?

Reply:
Thanks for the review! First of all: Is there any Documentation on those Portgroups other than the _resources directory? 

The normal method unfortunately did not work as the currently packaged Rust version, for whatever reason, assumes that everything inside the lib/ directory is a directory itself. So cargo looks for lib/README.md/Cargo.toml which obviously does not exist.

Reply:
fyi this is not an issue on the latest stable rust version

Reply:
Hey @fabius, I'll get back to you - I'd like to take a look at the `lib/REAME.md` issue you called out

Reply:
> Hey @fabius, I'll get back to you - I'd like to take a look at the `lib/REAME.md` issue you called out

@herbygillot Any news on this one? 

_I got a working Portfile for Rust 1.49/aarch64 if that helps you testing_